pypesto/optimize/util.py

```diff
@@ -102,6 +102,7 @@ def autosave(filename: str, result: Result,
         return None
     if filename == "Auto":
         time = datetime.datetime.now().strftime("%Y_%d_%m_%H_%M_%S")
-        filename = time+f"_{type}_result.hdf5"
-    write_result(result=result, overwrite=True,
-                 optimize=True, filename=filename)
+        filename = time+f"_{type}_result.h5"
+    # set the type to True and pass it on to write_result
+    to_save = {type: True}
```

Comment:
`type` shadows a built-in Python function so is unsuitable as a variable name. Usual Python convention is to rename with an underscore suffix, e.g. `type_`, or pick a different name, e.g. `result_type`).